File: mvps/dotnet/Dtos/ApiDtos.cs
namespace BeerVote.Dtos;

public sealed record HealthResponse(string Status, string Service, string Runtime);

public sealed record BeerItem(int Id, string Name);

public sealed record BeerListResponse(IReadOnlyList<BeerItem> Items);

public sealed record CreateUserRequest(string? Name);

public sealed record UserResponse(Guid Id, string Name, DateTimeOffset CreatedAt);

public sealed record UserListResponse(
    IReadOnlyList<UserResponse> Items,
    int Page,
    int PageSize,
    long Total,
    int TotalPages);

public sealed record CreateVoteRequest(Guid? UserId, int BeerId);

public sealed record VoteResponse(
    Guid Id,
    Guid UserId,
    int BeerId,
    string BeerName,
    DateTimeOffset VotedAt);

public sealed record ResultItem(int BeerId, string BeerName, long Votes, decimal Share);

public sealed record ResultsResponse(long TotalVotes, IReadOnlyList<ResultItem> Results);

public sealed record ErrorResponse(string Error, int Status);

File: java/src/main/java/com/meetup/beervote/dto/ApiDtos.java
package com.meetup.beervote.dto;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.List;
import java.util.UUID;

public final class ApiDtos {

    private ApiDtos() {
    }

    public record HealthResponse(String status, String service, String runtime) {
    }

    public record BeerItem(int id, String name) {
    }

    public record BeerListResponse(List<BeerItem> items) {
    }

    public record CreateUserRequest(String name) {
    }

    public record UserResponse(UUID id, String name, Instant createdAt) {
    }

    public record UserListResponse(
            List<UserResponse> items,
            int page,
            int pageSize,
            long total,
            int totalPages
    ) {
    }

    public record CreateVoteRequest(UUID userId, int beerId) {
    }

    public record VoteResponse(
            UUID id,
            UUID userId,
            int beerId,
            String beerName,
            Instant votedAt
    ) {
    }

    public record ResultItem(int beerId, String beerName, long votes, BigDecimal share) {
    }

    public record ResultsResponse(long totalVotes, List<ResultItem> results) {
    }

    public record ErrorResponse(String error, int status) {
    }
}
